crypto/fipsmodule/hkdf/hkdf.c

```diff
@@ -44,16 +45,26 @@ int HKDF_extract(uint8_t *out_key, size_t *out_len, const EVP_MD *digest,
                  size_t salt_len) {
   // https://tools.ietf.org/html/rfc5869#section-2.2
 
+  // We have to avoid the underlying HMAC services updating the indicator
```

Comment:
I think we will be able to insert the service indicator checks only in HKDF which does both Expansion and Extraction, and not in each step individually (and when we add it to the EVP interface, it will be only for the default setting of [EVP_PKEY_HKDEF_MODE_EXTRACT_AND_EXPAND] (https://commondatastorage.googleapis.com/chromium-boringssl-docs/kdf.h.html#EVP_PKEY_HKDEF_MODE_EXTRACT_AND_EXPAND))
I'm under that impression from the example ACVP tests in https://pages.nist.gov/ACVP/draft-hammett-acvp-kas-kdf-hkdf.html#name-example-test-vectors-json, which have a final output only.
See also [FIPS 140**-2** IG, Sec G.13, search for "KDA1 (Cert. #A25)"](https://csrc.nist.gov/csrc/media/projects/cryptographic-module-validation-program/documents/fips140-2/fips1402ig.pdf#%5B%7B%22num%22%3A94%2C%22gen%22%3A0%7D%2C%7B%22name%22%3A%22XYZ%22%7D%2C88%2C415%2C0%5D).

Reply:
The way CryptoAlg-1198 was written, it seemed that someone was using `HKDF_expand()` stand-alone, so I wanted to cover both the one-shot and the stand-alone entry points.

Reply:
It will not suffice to cover just the full HKDF(). 

It is possible to test and validate AWS-LC's HKDF_Expand() as a [KBKDF](https://pages.nist.gov/ACVP/draft-celi-acvp-kbkdf.html ) compliant with SP 800-108 (KDF in Feedback mode) by using `"fixedDataOrder"="after fixed data"` 

Just HKDF() will not suffice because it is not approved for key derivation from an imported key in SP 800-108.

Reply:
I'll add a link to this discussion in the HKDF ACVP story.